### PR TITLE
enhance: When describing an index, fetch the index info in batches

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #29313

### CODE REVIEW

File: internal/datacoord/index_meta.go [hunk 1]
```diff
@@ -854,3 +854,35 @@ func (m *indexMeta) GetMetasByNodeID(nodeID UniqueID) []*model.SegmentIndex {
 	}
 	return metas
 }
+
+func (m *indexMeta) getSegmentsIndexStats(collectionID UniqueID, segmentIDs []UniqueID) map[int64]map[int64]*indexpb.SegmentIndexState {
```
Comment (czs007) [inline internal/datacoord/index_meta.go:858]: better to rename as getSegmentsIndexStates
Stats is different from State

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**